File: dota-zone/dotazone/src/main/java/br/com/dotazone/view/activity/HeroProfileActivity.java
package br.com.dotazone.view.activity;

import android.animation.Animator;
import android.animation.Animator.AnimatorListener;
import android.animation.ObjectAnimator;
import android.animation.PropertyValuesHolder;
import android.animation.ValueAnimator;
import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import android.support.v4.widget.DrawerLayout;
import android.view.Gravity;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.google.android.gms.ads.AdView;
import com.google.android.gms.analytics.GoogleAnalytics;
import com.squareup.picasso.Picasso;

import br.com.dotazone.DotaZoneApplication;
import br.com.dotazone.DotaZoneBrain;
import br.com.dotazone.DotazoneMenu;
import br.com.dotazone.R;
import br.com.dotazone.model.entity.AdMobBanner;
import br.com.dotazone.model.entity.Hero;
import br.com.dotazone.model.entity.Skill;
import br.com.dotazone.model.util.UrlUtils;
import br.com.dotazone.view.components.FrameLayoutSkillBoard;
import br.com.dotazone.view.fragment.HeroSkillFragment;

public class HeroProfileActivity extends BaseActivity implements OnClickListener {

    private TextView mTextInt;
    private TextView mTextAgi;
    private TextView mTextStr;

    private TextView mTextAtk;
    private TextView mTextMove;
    private TextView mTextArmor;

    private ImageView mIconHero;
    private TextView mTextHeroName;
    private TextView mTextRoles;
    private Hero mHero = DotaZoneBrain.hero;
    private LinearLayout mLinearSkill;
    private ObjectAnimator mAnimator;
    private ImageView mIconHeroBio;
    private RelativeLayout mLinearBg;

    private DrawerLayout mDrawerLayout;
    private RelativeLayout mDrawerList;
    private DotazoneMenu mMenu;
    private AdView adView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.hero_profile_view);
        initComponents();
        ((DotaZoneApplication) getApplication()).getTracker(DotaZoneApplication.TrackerName.APP_TRACKER);

    }

    @Override
    protected void onStart() {
        super.onStart();

        GoogleAnalytics.getInstance(this).reportActivityStart(this);
    }

    @Override
    protected void onStop() {
        super.onStop();

        // Stop the analytics tracking
        GoogleAnalytics.getInstance(this).reportActivityStop(this);
    }

    @Override
    public void initComponents() {

        mTextAgi = (TextView) findViewById(R.id.hero_text_attribute_agi);
        mTextStr = (TextView) findViewById(R.id.hero_text_attribute_str);
        mTextInt = (TextView) findViewById(R.id.hero_text_attribute_int);
        mTextAtk = (TextView) findViewById(R.id.hero_text_attribute_atk);
        mTextMove = (TextView) findViewById(R.id.hero_text_attribute_move);
        mTextArmor = (TextView) findViewById(R.id.hero_text_attribute_armor);
        mIconHero = (ImageView) findViewById(R.id.hero_icon_avatar);
        mTextHeroName = (TextView) findViewById(R.id.hero_text_name);
        mTextRoles = (TextView) findViewById(R.id.hero_text_roles);
        mIconHeroBio = (ImageView) findViewById(R.id.hero_icon_bio);
        mIconHeroBio.setOnClickListener(this);
        mLinearBg = (RelativeLayout) findViewById(R.id.hero_linear_bg);

        int idImageBackground = getResources().getIdentifier(mHero.getIdString() + "_bg", "drawable", getPackageName());
        mLinearBg.setBackgroundResource(idImageBackground);

        String nameHero = mHero.getName().replace("_", " ");
        mTextHeroName.setText(nameHero);
        mTextRoles.setText(mHero.getAbilites().getDroles());
        Typeface font = Typeface.createFromAsset(getAssets(), "Roboto-Thin.ttf");
        mTextHeroName.setTypeface(font);

        mTextAgi.setText(mHero.getAbilites().getAgi()[0]);
        mTextInt.setText(mHero.getAbilites().getInte()[0]);
        mTextStr.setText(mHero.getAbilites().getStr()[0]);
        mTextStr.setText(mHero.getAbilites().getStr()[0]);

        mTextAtk.setText(mHero.getAbilites().getDmg()[0]);
        mTextMove.setText(mHero.getAbilites().getMs());
        mTextArmor.setText(mHero.getAbilites().getArmor());

        int idImage = getResources().getIdentifier(mHero.getIdString() + "_avatar", "drawable", getPackageName());

        if (idImage != 0) {

            Picasso.with(this).load(idImage).into(mIconHero);
            mIconHero.setImageResource(idImage);
        }

        mLinearSkill = (LinearLayout) findViewById(R.id.hero_skill_layout);
        loandingSkillsHero();

        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
        mDrawerList = (RelativeLayout) findViewById(R.id.list_slidermenu);
        mMenu = new DotazoneMenu(this, mDrawerLayout, mDrawerList);
        mMenu.checkHeroMenu();

        adView = (AdView) findViewById(R.id.hero_skill_admob);
        new AdMobBanner().createBanner(this, adView, DotaZoneBrain.isPremium);
    }

    @Override
    public void onPause() {
        adView.pause();
        super.onPause();
    }

    @Override
    public void onResume() {
        super.onResume();
        adView.resume();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        adView.destroy();
    }

    private FrameLayoutSkillBoard createFrameLayoutBrightness(FrameLayout mainFrame, int idFrame) {

        FrameLayoutSkillBoard frameBrightness = new FrameLayoutSkillBoard(this);
        frameBrightness.setId(idFrame);
        mainFrame.addView(frameBrightness);
        FrameLayout.LayoutParams paramBright = (FrameLayout.LayoutParams) new FrameLayout.LayoutParams(FrameLayout.LayoutParams.WRAP_CONTENT,
                FrameLayout.LayoutParams.WRAP_CONTENT);
        paramBright.height = UrlUtils.convertDpToPixel(87, getResources());
        paramBright.width = UrlUtils.convertDpToPixel(87, getResources());
        frameBrightness.setLayoutParams(paramBright);

        return frameBrightness;
    }

    private FrameLayout createMainFrameSkill() {

        FrameLayout mainFrame = new FrameLayout(this);
        FrameLayout.LayoutParams paramMainFrame = (FrameLayout.LayoutParams) new FrameLayout.LayoutParams(FrameLayout.LayoutParams.WRAP_CONTENT,
                FrameLayout.LayoutParams.WRAP_CONTENT);
        paramMainFrame.height = UrlUtils.convertDpToPixel(87, getResources());
        paramMainFrame.width = UrlUtils.convertDpToPixel(87, getResources());
        paramMainFrame.gravity = Gravity.CENTER_HORIZONTAL;
        mainFrame.setLayoutParams(paramMainFrame);

        return mainFrame;
    }

    private ImageView createSkillIcon(FrameLayout mainFrame, int idImage) {

        ImageView skillImage = new ImageView(this);
        skillImage.setImageResource(idImage);
        mainFrame.addView(skillImage);
        FrameLayout.LayoutParams teste = (FrameLayout.LayoutParams) skillImage.getLayoutParams();
        teste.height = UrlUtils.convertDpToPixel(64, getResources());
        teste.width = UrlUtils.convertDpToPixel(64, getResources());
        teste.leftMargin = UrlUtils.convertDpToPixel(11, getResources());
        teste.topMargin = UrlUtils.convertDpToPixel(11, getResources());
        skillImage.setLayoutParams(teste);

        return skillImage;
    }

    private void loandingSkillsHero() {

        final int total = mHero.getSkills().size();

        for (int i = 0; i < total; i++) {

            final Skill skill = mHero.getSkills().get(i);
            FrameLayout mainFrame = createMainFrameSkill();
            final FrameLayoutSkillBoard frameBrightness = createFrameLayoutBrightness(mainFrame, i);
            final int idImage = getResources().getIdentifier(skill.getIdString(), "drawable", getPackageName());
            createSkillIcon(mainFrame, idImage);

            frameBrightness.setOnClickListener(new OnClickListener() {

                @Override
                public void onClick(View v) {

                    frameBrightness.changeBackgroud();
                    animationFadeIn(frameBrightness, false);
                    for (int i = 0; i < total; i++) {
                        FrameLayoutSkillBoard lastFramLayout = (FrameLayoutSkillBoard) findViewById(i);

                        if (frameBrightness.getId() != lastFramLayout.getId())
                            lastFramLayout.removeBackground();
                    }

                    HeroSkillFragment fragment1 = HeroSkillFragment.newInstance(skill);
                    getSupportFragmentManager().beginTransaction().replace(R.id.hero_description_skill_layout, fragment1).commit();
                }
            });

            // default skill in zero position.
            if (i == 0) {

                frameBrightness.changeBackgroud();
                animationFadeIn(frameBrightness, false);
                HeroSkillFragment fragment1 = HeroSkillFragment.newInstance(skill);
                getSupportFragmentManager().beginTransaction().replace(R.id.hero_description_skill_layout, fragment1).commit();
            }

            mLinearSkill.addView(mainFrame);

        }

    }

    private void animationFadeIn(final FrameLayoutSkillBoard frameBrightness, final boolean cancel) {

        frameBrightness.setAlpha(0);
        PropertyValuesHolder propValueAlpha = PropertyValuesHolder.ofFloat("alpha", (float) 0.5, 1);
        mAnimator = ObjectAnimator.ofPropertyValuesHolder(frameBrightness, propValueAlpha);
        mAnimator.setDuration(500);
        mAnimator.setRepeatCount(ValueAnimator.INFINITE);
        mAnimator.setRepeatMode(ValueAnimator.REVERSE);
        mAnimator.start();
        mAnimator.addListener(new AnimatorListener() {

            @Override
            public void onAnimationStart(Animator animation) {

            }

            @Override
            public void onAnimationRepeat(Animator animation) {

                if (cancel)
                    mAnimator.cancel();
            }

            @Override
            public void onAnimationEnd(Animator animation) {

            }

            @Override
            public void onAnimationCancel(Animator animation) {

            }
        });

    }

    @Override
    public void onClick(View arg0) {

        startActivity(new Intent(this, HeroBiographActivity.class));

    }

    @Override
    public void setActionErrorOk() {
        finish();

    }

    @Override
    public void setActionErrorCancel() {
        finish();

    }

}

File: dota-zone/app/src/main/java/com/br/dotazone/view/activity/HeroProfileActivity.kt
package com.br.dotazone.view.activity

import android.animation.Animator
import android.animation.ObjectAnimator
import android.animation.PropertyValuesHolder
import android.animation.ValueAnimator
import android.content.Intent
import android.graphics.Typeface
import android.os.Bundle
import android.view.Gravity
import android.view.View
import android.widget.*
import androidx.drawerlayout.widget.DrawerLayout
import com.br.dotazone.DotaZoneApplication
import com.br.dotazone.DotaZoneBrain.hero
import com.br.dotazone.DotaZoneBrain.isPremium
import com.br.dotazone.R
import com.br.dotazone.model.entity.AdMobBanner
import com.br.dotazone.model.util.UrlUtils.Companion.convertDpToPixel
import com.br.dotazone.view.components.FrameLayoutSkillBoard
import com.br.dotazone.view.fragment.HeroSkillFragment.Companion.newInstance
import com.com.dotazone.DotazoneMenu
import com.google.android.gms.ads.AdView
import com.google.android.gms.analytics.GoogleAnalytics


class HeroProfileActivity : BaseActivity(), View.OnClickListener {
	private var mTextInt: TextView? = null
	private var mTextAgi: TextView? = null
	private var mTextStr: TextView? = null
	private var mTextAtk: TextView? = null
	private var mTextMove: TextView? = null
	private var mTextArmor: TextView? = null
	private var mIconHero: ImageView? = null
	private var mTextHeroName: TextView? = null
	private var mTextRoles: TextView? = null
	private val mHero = hero
	private var mLinearSkill: LinearLayout? = null
	private var mAnimator: ObjectAnimator? = null
	private var mIconHeroBio: ImageView? = null
	private var mLinearBg: RelativeLayout? = null
	private var mDrawerLayout: DrawerLayout? = null
	private var mDrawerList: RelativeLayout? = null
	private var mMenu: DotazoneMenu? = null
	private var adView: AdView? = null
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.hero_profile_view)
		initComponents()
	}

	override fun onStart() {
		super.onStart()
		GoogleAnalytics.getInstance(this).reportActivityStart(this)
	}

	override fun onStop() {
		super.onStop()

		// Stop the analytics tracking
		GoogleAnalytics.getInstance(this).reportActivityStop(this)
	}

	override fun initComponents() {
		mTextAgi = findViewById<View>(R.id.heroTextAttributeAgi) as TextView
		mTextStr = findViewById<View>(R.id.heroTextAttributeStr) as TextView
		mTextInt = findViewById<View>(R.id.heroTextAttributeInt) as TextView
		mTextAtk = findViewById<View>(R.id.heroTextAttributeAtk) as TextView
		mTextMove = findViewById<View>(R.id.heroTextAttributeMove) as TextView
		mTextArmor = findViewById<View>(R.id.heroTextAttributeArmor) as TextView
		mIconHero = findViewById<View>(R.id.hero_icon_avatar) as ImageView
		mTextHeroName = findViewById<View>(R.id.heroTextName) as TextView
		mTextRoles = findViewById<View>(R.id.hero_text_roles) as TextView
		mIconHeroBio = findViewById<View>(R.id.hero_icon_bio) as ImageView
		mIconHeroBio!!.setOnClickListener(this)
		mLinearBg = findViewById<View>(R.id.hero_linear_bg) as RelativeLayout
		val idImageBackground = resources.getIdentifier(mHero!!.idString + "_bg", "drawable", packageName)
		mLinearBg!!.setBackgroundResource(idImageBackground)
		val nameHero = mHero.name?.replace("_", " ")
		mTextHeroName!!.text = nameHero
		mTextRoles!!.text = mHero.abilites?.droles
		val font = Typeface.createFromAsset(assets, "Roboto-Thin.ttf")
		mTextHeroName!!.setTypeface(font)
		mTextAgi!!.text = mHero.abilites?.agi?.get(0) ?: ""
		mTextInt!!.text = mHero.abilites?.inte?.get(0) ?: ""
		mTextStr!!.text = mHero.abilites?.str?.get(0) ?: ""
		mTextStr!!.text = mHero.abilites?.str?.get(0) ?: ""
		mTextAtk!!.text = mHero.abilites?.dmg?.get(0) ?: ""
		mTextMove!!.text = mHero.abilites?.ms
		mTextArmor!!.text = mHero.abilites?.armor
		val idImage = resources.getIdentifier(mHero.idString + "_avatar", "drawable", packageName)
		if (idImage != 0) {
			mIconHero!!.setImageResource(idImage)
		}
		mLinearSkill = findViewById<View>(R.id.hero_skill_layout) as LinearLayout
		loandingSkillsHero()
		mDrawerLayout = findViewById<View>(R.id.drawer_layout) as DrawerLayout
		mDrawerList = findViewById<View>(R.id.listSliderMenu) as RelativeLayout
		mMenu = DotazoneMenu(this, mDrawerLayout, mDrawerList)
		mMenu!!.checkHeroMenu()
		adView = findViewById<View>(R.id.hero_skill_admob) as AdView
		AdMobBanner().createBanner(this, adView!!, isPremium)
	}

	public override fun onPause() {
		adView!!.pause()
		super.onPause()
	}

	public override fun onResume() {
		super.onResume()
		adView!!.resume()
	}

	public override fun onDestroy() {
		super.onDestroy()
		adView!!.destroy()
	}

	private fun createFrameLayoutBrightness(mainFrame: FrameLayout, idFrame: Int): FrameLayoutSkillBoard {
		val frameBrightness = FrameLayoutSkillBoard(this)
		frameBrightness.id = idFrame
		mainFrame.addView(frameBrightness)
		val paramBright = FrameLayout.LayoutParams(FrameLayout.LayoutParams.WRAP_CONTENT,
				FrameLayout.LayoutParams.WRAP_CONTENT)
		paramBright.height = convertDpToPixel(87f, resources)
		paramBright.width = convertDpToPixel(87f, resources)
		frameBrightness.layoutParams = paramBright
		return frameBrightness
	}

	private fun createMainFrameSkill(): FrameLayout {
		val mainFrame = FrameLayout(this)
		val paramMainFrame = FrameLayout.LayoutParams(FrameLayout.LayoutParams.WRAP_CONTENT,
				FrameLayout.LayoutParams.WRAP_CONTENT)
		paramMainFrame.height = convertDpToPixel(87f, resources)
		paramMainFrame.width = convertDpToPixel(87f, resources)
		paramMainFrame.gravity = Gravity.CENTER_HORIZONTAL
		mainFrame.layoutParams = paramMainFrame
		return mainFrame
	}

	private fun createSkillIcon(mainFrame: FrameLayout, idImage: Int): ImageView {
		val skillImage = ImageView(this)
		skillImage.setImageResource(idImage)
		mainFrame.addView(skillImage)
		val teste = skillImage.layoutParams as FrameLayout.LayoutParams
		teste.height = convertDpToPixel(64f, resources)
		teste.width = convertDpToPixel(64f, resources)
		teste.leftMargin = convertDpToPixel(11f, resources)
		teste.topMargin = convertDpToPixel(11f, resources)
		skillImage.layoutParams = teste
		return skillImage
	}

	private fun loandingSkillsHero() {
		val total = mHero!!.skills.size
		for (i in 0 until total) {
			val skill = mHero.skills[i]
			val mainFrame = createMainFrameSkill()
			val frameBrightness = createFrameLayoutBrightness(mainFrame, i)
			val idImage = resources.getIdentifier(skill.idString, "drawable", packageName)
			createSkillIcon(mainFrame, idImage)
			frameBrightness.setOnClickListener {
				frameBrightness.changeBackgroud()
				animationFadeIn(frameBrightness, false)
				for (i in 0 until total) {
					val lastFramLayout = findViewById<View>(i) as FrameLayoutSkillBoard
					if (frameBrightness.id != lastFramLayout.id) lastFramLayout.removeBackground()
				}
				val fragment1 = newInstance(skill)
				supportFragmentManager.beginTransaction().replace(R.id.hero_description_skill_layout, fragment1).commit()
			}

			// default skill in zero position.
			if (i == 0) {
				frameBrightness.changeBackgroud()
				animationFadeIn(frameBrightness, false)
				val fragment1 = newInstance(skill)
				supportFragmentManager.beginTransaction().replace(R.id.hero_description_skill_layout, fragment1).commit()
			}
			mLinearSkill!!.addView(mainFrame)
		}
	}

	private fun animationFadeIn(frameBrightness: FrameLayoutSkillBoard, cancel: Boolean) {
		frameBrightness.alpha = 0f
		val propValueAlpha = PropertyValuesHolder.ofFloat("alpha", 0.5.toFloat(), 1f)
		mAnimator = ObjectAnimator.ofPropertyValuesHolder(frameBrightness, propValueAlpha)
		mAnimator!!.duration = 500
		mAnimator!!.repeatCount = ValueAnimator.INFINITE
		mAnimator!!.repeatMode = ValueAnimator.REVERSE
		mAnimator!!.start()
		mAnimator!!.addListener(object : Animator.AnimatorListener {
			override fun onAnimationStart(animation: Animator) {}
			override fun onAnimationRepeat(animation: Animator) {
				if (cancel) mAnimator!!.cancel()
			}

			override fun onAnimationEnd(animation: Animator) {}
			override fun onAnimationCancel(animation: Animator) {}
		})
	}

	override fun onClick(arg0: View) {
		startActivity(Intent(this, HeroBiographActivity::class.java))
	}

	override fun setActionErrorOk() {
		finish()
	}

	override fun setActionErrorCancel() {
		finish()
	}
}

